### PR TITLE
[Docs] Visualizing the output of data pipeline as snippet

### PR DESCRIPTION
**Pull request by marijnl**
## Motivation

While I was creating a custom data pipeline it was not clear how to visualize the output

## Modification

Added a snippet to the data pipeline tutorial that helps the user visualize the output of his data pipeline

### CODE REVIEW

**Comment by marijnl** [comment]
#5225

File: docs/tutorials/data_pipeline.md [hunk 1]
```diff
@@ -146,28 +146,29 @@ For each operation, we list the related dict fields that are added/updated/remov
 
 ## Extend and use custom pipelines
 
-1. Write a new pipeline in any file, e.g., `my_pipeline.py`. It takes a dict as input and return a dict.
+1. Write a new pipeline e.g., `my_pipeline.py`. It takes a dict as input and returns a dict.
```
Comment (ZwwWayne) [inline docs/tutorials/data_pipeline.md:149]: suggest to rephrase it as "Write a new pipeline in a file, e.g., in `my_pipeline.py`."

**Comment by ZwwWayne** [comment]
Hi @marijnl ,
Thanks for your kind contribution. It seems this PR only has few issues with the writing. Could you fix them?

File: docs/tutorials/data_pipeline.md [hunk 2]
```diff
@@ -146,28 +146,29 @@ For each operation, we list the related dict fields that are added/updated/remov
...
 
     ```python
+    import random
     from mmdet.datasets import PIPELINES
 
     @PIPELINES.register_module()
     class MyTransform:
+        def __init__(self, p=.5):
```
Comment (BIGWangYuDong) [inline docs/tutorials/data_pipeline.md:157]: suggest setting `p=0.5`, and add you can add docstring.
An example:
https://github.com/open-mmlab/mmdetection/blob/682f03db0c0516324ec4e1e5d97e1d93c0a2485a/mmdet/datasets/pipelines/transforms.py#L479-L482

File: docs/tutorials/data_pipeline.md [hunk 3]
```diff
@@ -146,28 +146,29 @@ For each operation, we list the related dict fields that are added/updated/remov
...
-2. Import the new class.
+2. Import and use the pipeline in your config file. 
+   Make sure the import is relative to where your train script is located.
 
     ```python
-    from .my_pipeline import MyTransform
-    ```
-
-3. Use it in config files.
+    custom_imports = dict(imports=['my_pipeline'], allow_failed_imports=False)
```
Comment (BIGWangYuDong) [inline docs/tutorials/data_pipeline.md:170]: maybe can add an annotation, `the path to my_pipeline, e.g. mmdet.datasets.pipelines.my_pipeline`

**Review by BIGWangYuDong — COMMENTED** [review]
The visualize of the output aug pipeline seems to work, I have already tried this code.
- Need to change the Chinese version docs too.

**Comment by BIGWangYuDong** [comment]
Already add Chinese version of datapipeline

File: docs_zh-CN/tutorials/data_pipeline.md [hunk 4]
```diff
@@ -170,8 +177,29 @@ test_pipeline = [
...
         dict(type='Normalize', **img_norm_cfg),
         dict(type='Pad', size_divisor=32),
-        dict(type='MyTransform'),
+        dict(type='MyTransform', p=0.2),
         dict(type='DefaultFormatBundle'),
         dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
     ]
     ```
+
+3. 可视化增强数据处理流程的结果
```
Comment (hhaAndroid) [inline docs_zh-CN/tutorials/data_pipeline.md:186]: Visual pipeline can use `tools/misc/browse_dataset.py`

File: docs_zh-CN/tutorials/data_pipeline.md [hunk 5]
```diff
@@ -139,28 +139,35 @@ test_pipeline = [
...
 
         def __call__(self, results):
-            results['dummy'] = True
+            if random.random() > self.p:
+                results['dummy'] = True
             return results
     ```
 
-2. 导入这个新类，确保程序启动时它会被注册进 PIPELINES：
+2. 在配置文件里调用并使用你写的数据处理流程，需要确保在你的训练脚本中调用了准确的位置：
```
Comment (hhaAndroid) [inline docs_zh-CN/tutorials/data_pipeline.md:166]: 需要确保你的训练脚本能够正确导入新增模块

File: docs_zh-CN/tutorials/data_pipeline.md [hunk 6]
```diff
@@ -170,8 +177,14 @@ test_pipeline = [
         dict(type='RandomFlip', flip_ratio=0.5),
         dict(type='Normalize', **img_norm_cfg),
         dict(type='Pad', size_divisor=32),
-        dict(type='MyTransform'),
+        dict(type='MyTransform', p=0.2),
         dict(type='DefaultFormatBundle'),
         dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
     ]
     ```
+
```
Comment (hhaAndroid) [inline docs_zh-CN/tutorials/data_pipeline.md:185]: 可视化数据增强处理流程的结果

File: docs_zh-CN/tutorials/data_pipeline.md [hunk 7]
```diff
@@ -170,8 +177,14 @@ test_pipeline = [
...
+3. 可视化增强数据处理流程的结果
+
+   如果想要可视化增强处理数据流程的结果，可以使用 `tools/misc/browse_dataset.py` 直观
```
Comment (hhaAndroid) [inline docs_zh-CN/tutorials/data_pipeline.md:188]: 数据增强处理流程的结果

**Comment by hhaAndroid** [comment]
@marijnl Please merge the master branch so that all CI can pass.

**Comment by BIGWangYuDong** [comment]
Ping @hhaAndroid, done

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**